api/team_members.go

```diff
@@ -64,26 +100,142 @@ func (svc *TeamMembersService) List() (res TeamMembersResponse, err error) {
+	}
+	if guid == "" {
+		return errors.New("please specify a guid")
+	}
 	return svc.client.RequestDecoder("DELETE", fmt.Sprintf(apiV2TeamMembersFromGUID, guid), nil, nil)
 }
 
-// Update updates a single team member with the corresponding guid
+// Update updates a single team member at the account-level with the corresponding guid
 func (svc *TeamMembersService) Update(tm TeamMember) (res TeamMemberResponse, err error) {
+	if svc.client.OrgAccess() {
+		return res, errors.New("client configured to manage org-level datasets, use UpdateOrg()")
+	}
 	if tm.UserGuid == "" {
 		err = errors.New("please specify a guid")
 		return
 	}
-	err = svc.client.RequestEncoderDecoder("PATCH", fmt.Sprintf(apiV2TeamMembersFromGUID, tm.UserGuid), tm, &res)
+	userGuid := tm.UserGuid
+	// Omit userGuid for patch requests
+	tm.UserGuid = ""
+	// Omit userName for patch requests as it cannot be modified
+	tm.UserName = ""
+	err = svc.client.RequestEncoderDecoder("PATCH", fmt.Sprintf(apiV2TeamMembersFromGUID, userGuid), tm, &res)
+	return
+}
+
+// UpdateOrg updates a single team member at the org-level with the corresponding username
+func (svc *TeamMembersService) UpdateOrg(tm TeamMemberOrg) (res TeamMemberOrgResponse, err error) {
+	if !svc.client.OrgAccess() {
+		return res, errors.New("client configured to manage account-level datasets, use Update()")
+	}
+	if tm.UserName == "" {
+		err = errors.New("please specify a username")
+		return
+	}
+	var userGuid string
+	tms, errSearch := svc.SearchUsername(tm.UserName)
+	if errSearch != nil || len(tms.Data) == 0 {
+		err = errors.Wrap(err, "unable to find user with specified username")
+		return
+	}
+	userGuid = tms.Data[0].UserGuid
+	// Omit UserGuid from the patch body as it cannot be modified
+	tm.UserGuid = ""
+	// Omit userEnabled field from the patch body as it cannot be modified
+	tm.UserEnabled = 0
+	// Omit userName field from the patch body as it cannot be modified
+	tm.UserName = ""
+	// Omit Company field from the patch body as it cannot be modified
+	tm.Props.Company = ""
+	patchBody, patchErr := jsonReader(tm)
+	if patchErr != nil {
+		err = errors.Wrap(patchErr, "unable to encode team member as JSON")
+		return
+	}
+
+	req, patchReqErr := svc.client.NewRequest("PATCH", fmt.Sprintf(apiV2TeamMembersFromGUID, userGuid), patchBody)
+	if patchReqErr != nil {
+		err = errors.Wrap(patchReqErr, "unable to create patch request")
+		return
+	}
+
+	response, doErr := svc.client.Do(req)
+	if doErr != nil {
+		err = errors.Wrap(doErr, "unable to perform patch request")
+		return
+	}
+
+	defer response.Body.Close()
+
+	body, bodyErr := ioutil.ReadAll(response.Body)
+
+	if bodyErr != nil {
+		err = errors.Wrap(bodyErr, "unable to read json body")
+		return
+	}
+
+	unmarshalErr := json.Unmarshal(body, &res)
+	if unmarshalErr != nil {
+		err = errors.Wrap(unmarshalErr, "unable to deserialize response body into team member struct")
+		return
+	}
+	return
```

Comment:
Can this be replaced by:
```suggestion
	err = svc.client.RequestEncoderDecoder("PATCH", fmt.Sprintf(apiV2TeamMembersFromGUID, userGuid), tm, &res)
```

or are we doing something special?